aten/src/ATen/native/NNPACK.cpp

```diff
@@ -243,16 +252,35 @@ Tensor _nnpack_spatial_convolution(
                                             .left = (size_t)padding[1]};
   const struct nnp_size kernel_size = {.width = (size_t)weight.size(3),
                                        .height = (size_t)weight.size(2)};
+  const nnp_size output_subsample = {.width = stride[1], .height = stride[0]};
 
+  const auto input_ = input.contiguous();
   // If we don't have a defined bias Tensor, we need to create one filled with
   // zeroes
-  auto bias_ =
+  const auto bias_ =
       bias.defined() ? bias : at::zeros({weight.size(0)}, input.options());
 
   // Note: we assume that the output is shaped correctly, probably should add an
   // assert
-  auto input_ = input.contiguous();
   auto batched = [&]() -> nnp_status {
+    if ((output_subsample.width != 1) || (output_subsample.height != 1)) {
+      nnp_convolution_output__reference(
+          batch_size,
+          input_channels,
+          output_channels,
+          input_size,
+          input_padding,
+          kernel_size,
+          output_subsample,
+          input_.data_ptr<float>(),
+          weight.data_ptr<float>(),
+          bias_.data_ptr<float>(),
+          output.data_ptr<float>(),
+          nnpack_threadpool());
```

Comment:
NNPACK does not support strided convolutions on batches (used during training,) hence we fallback to the slower reference implementation here.